src/ru/nsu/ccfit/zuev/osu/game/cursor/flashlight/MainFlashLightSprite.java

```diff
@@ -0,0 +1,43 @@
+package ru.nsu.ccfit.zuev.osu.game.cursor.flashlight;
+
+import org.anddev.andengine.entity.modifier.ScaleModifier;
+import org.anddev.andengine.opengl.texture.region.TextureRegion;
+
+
+public class MainFlashLightSprite extends FlashlightAreaSizedSprite {
+    public static final float BASE_SIZE = 8f;
+    public static final int TEXTURE_WIDTH = 1024;
+    public static final int TEXTURE_HEIGHT = TEXTURE_WIDTH / 2;
+    public final float AREA_SHRINK_FADE_DURATION = 0.8f;
+    public float currentSize = BASE_SIZE;
+
+    public MainFlashLightSprite(TextureRegion pTextureRegion) {
+        super(pTextureRegion);
+    }
+
+    public void update(int combo, boolean isShowing) {
+        this.setVisible(isShowing);
+        this.handleAreaShrinking(combo);
+    }
+
+    public void shrinkArea(float fromScale, float toScale) {
+        this.registerEntityModifier(new ScaleModifier(AREA_SHRINK_FADE_DURATION, fromScale, toScale));
+    }
```

Comment:
Ditto.

`fromScale` shouldn't even exist; it should just use the current scale instead of having to be specified.

```suggestion
    public void changeArea(float toScale) {
        this.registerEntityModifier(new ScaleModifier(AREA_CHANGE_FADE_DURATION, currentSize, toScale));
    }
```